src/components/Cart/CartAddDialog.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <dialog-template :title="title" @close="close">
+    <div :class="$style.container">
+      <form @submit.prevent="submit">
+        <owner-selector
+          v-if="item.type !== ItemType.equipment"
+          v-model="ownerName"
+          :owner-details="ownerDetails"
+          :class="$style.input"
+        />
+        <input-number
+          v-model="count"
+          :class="$style.input"
+          label="個数"
+          :max="maxCount"
+          :min="isEdit ? 0 : 1"
+        />
+        <wide-icon-button
+          :icon="button.icon"
+          :label="button.label"
+          :variant="button.variant"
+          type="submit"
+          :class="$style.button"
+          :disabled="maxCount === 0"
+        />
```

Comment:
```suggestion
        <wide-icon-button
          v-if="maxCount.value === 0"
          icon="mdi:cancel"
          label="在庫がありません"
          variant="caution"
          type="submit"
          :class="$style.button"
          disabled
        />
        <template v-else-if="isEdit">
          <wide-icon-button
            v-show="count === 0"
            icon="mdi:arrow-right-bold-circle"
            label="変更する"
            type="submit"
            :class="$style.button"
          />
          <wide-icon-button
            v-show="count !== 0"
            icon="mdi:arrow-right-bold-circle"
            label="削除する"
            variant="caution"
            type="submit"
            :class="$style.button"
          />
        </template>
        <wide-icon-button
          v-else
          icon="mdi:cart"
          label="カートに入れる"
          type="submit"
          :class="$style.button"
        />
```
このほうが読みやすいと思う
あと`btn`ってpropsわたすんじゃなくてここも含めて`<slot />`にしたcomponentをつくってそれをこれとCartAddDialogWithContinueで使うほうがいいと思う

なんならisEditかどうかもコンポーネント切ったほうがいいと思う

Reply:
結構考えたんだけどどういうことかわからなかった
最初の「ここ」はどのボタンを出すかのコンポーネントのこと(以降`SubmitButton`とする)を指してて、
「も含めた`<slot />`にする」っていうのは`CartAddDialog`を
```
  <dialog-template>
      <form>
        <owner-selector  />
        <input-number />
        <slot name="button">
          <submit-button />
        </slot>
      </form>
  </dialog-template>
```
みたいにするっていうこと？(fallbackにするかはおいといて)
「それをこれとCartAddDialogWithContinueで使う」っていうのは「それ」は上みたいな構成になった`CartAddDialog`、「これ」は`SubmitButton`のこと？
`CartAddDialogWithContinue`は
```
  <cart-add-dialog>
    <template #button>
      <div :class="$style.continue">
        <wide-icon-button />
        <div :class="$style.or">または</div>
        <wide-icon-button />
        <div :class="$style.description">説明</div>
      </div>
    </template>
  </cart-add-dialog>
```
こうするっていうこと？
この場合`v-slot:default`で関数を受け渡さないとダメそうだから違うと思うんだけどもうちょっと詳しく書いてほしい:pray:ｽﾏｿ

Reply:
意図してたのはそんな感じだけど、まぁ普通に愚直にきればいっかになった https://github.com/traPtitech/booQ-UI/pull/570#discussion_r641525671